doc/design/numeric_typed_column_clause.md

```diff
@@ -0,0 +1,164 @@
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(DENSE(c1, 10)), NUMERIC(DENSE(c2, 10)) -- 10 is the length of the decoded dense vector
+        INTO result_table;
+        ```
+    - If the decoded vector is sparse, there is only one way to write the `TO TRAIN` SQL statement.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(SPARSE(c1, 10)), NUMERIC(SPARSE(c2, 10)) -- 10 is the dense length of the decoded sparse vector
+        INTO result_table;
+        ```
+
+There are some problems in the current design:
+
+- Both `NUMERIC` and `DENSE/SPARSE` have the `length/shape` parameter. 
+For example, the expression `COLUMN NUMERIC(DENSE(c1, 10), 20)` is confused to users. 
+Users do not know whether the length of `c1` is 10 (the `length/shape` parameter of `DENSE`) or 20 (the `length/shape` parameter of `NUMERIC`).
+- `NUMERIC` is unnecessary in the `COLUMN` clause.
+    - For the column whose cell value is a single number, we can write `COLUMN c1`, `COLUMN NUMERIC(c1)` or `COLUMN NUMERIC(DENSE(c1))`, and `COLUMN c1` is simpler.
+    - For the column whose cell value is a string that is encoded from a numeric vector:
+        - For dense data, we can write simpler expression `COLUMN DENSE(c1, length)` instead of `COLUMN NUMERIC(DENSE(c1, length))` .
+        - For sparse data, we can write simpler expression `COLUMN SPARSE(c1, length)` instead of `COLUMN NUMERIC(SPARSE(c1, length))` .
+        
+In conclusion, we do not need the `NUMERIC` in the `COLUMN` clause. 
+We can load the data from the numeric column just using `DENSE` and `SPARSE` in the `COLUMN` clause.
+
+## Design
+
+### Changes on the APIs
+
+In the new design, we would remove `NUMERIC` in the `COLUMN` clause. We can use `DENSE/SPARSE` directly in the SQL statements.
+
+For the column whose cell value is a single number, there would be still 3 equivalent ways to write the `TO TRAIN` SQL statement.
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+LABEL label
+INTO result_table;
+```
+or
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+COLUMN c1, c2
+LABEL label
+INTO result_table;
+```
+or
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+COLUMN DENSE(c1), DENSE(c2)
+LABEL label
+INTO result_table;
+```
+
+For the column whose cell value is a string that is encoded from a numeric vector, there would be only one way to write the `TO TRAIN` SQL statement.
+- For dense data, the SQL statement would be:
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+LABEL label
+COLUMN DENSE(c1, 10), DENSE(c2, 10)
+INTO result_table;
+```
+- For sparse data, the SQL statement would be:
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+LABEL label
+COLUMN SPARSE(c1, 10), SPARSE(c2, 10)
```

Comment:
Also add examples of how to embed SPARSE to other columns like `EMBEDDING(SPARSE(...))`?

Reply:
Done.